mypy version bump 1.12.0

dirrao opened the pull request:
mypy version bump 1.12.0

dirrao committed:
mypy version bump 1.10.0

dirrao committed:
Merge branch 'main' into mypy_version_bump

dirrao committed:
mypy version bump 1.10.0

dirrao committed:
Merge branch 'main' into mypy_version_bump

uranusjr reviewed (changes requested):
We’re hitting https://github.com/python/typeshed/issues/10653 (giving this a red cross to avoid accidental merge)

dirrao committed:
mypy version bump 1.10.0

dirrao committed:
mypy version bump 1.11.0

dirrao committed:
replaced keyword argument with positional argument

dirrao committed:
Merge branch 'main' into mypy_version_bump

dirrao committed:
Merge branch 'main' into mypy_version_bump

uranusjr committed:
Add ignore to cast() call

Mypy is weirdly complaining it can't convert a callable to a
TaskDecoratorCollection since the type is not callable. This is wrong
in both ways---TaskDecoratorCollection *is* totally callable, and isn't
cast() supposed to not care anyway?

I don't see there's a viable way to work around this without
significantly impacting typing in other ways, so an ignore marker seems
to be the most reasonable option. The cast() call is kept because Mypy
might fix this ins the future, and when it does, it should report the
marker as redundant and prompt us to remove it.

dirrao committed:
mypy version bump 1.11.2

dirrao committed:
Merge branch 'main' into mypy_version_bump

dirrao committed:
Merge branch 'main' into mypy_version_bump

dirrao committed:
Merge branch 'main' into mypy_version_bump

dirrao committed:
mypy version bump 1.12.0

dirrao committed:
Merge branch 'main' into mypy_version_bump

hauntsaninja commented on hatch_build.py:
```diff
@@ -215,7 +215,7 @@
         # TODO: upgrade to newer versions of MyPy continuously as they are released
         # Make sure to upgrade the mypy version in update-common-sql-api-stubs in .pre-commit-config.yaml
```
should be updated in update-common-sql-api-stubs as per comment?

hauntsaninja reviewed (commented):
Looks like there are a few things from CI that need changes
